source-code/core/src/lint/linter.ts

```diff
@@ -13,46 +13,39 @@ import type {
 const getResourceForLanguage = (resources: Resource[], language: string) =>
 	resources.find(({ languageTag }) => languageTag.name === language)
 
-export const getLintRulesFromConfig = (config: Config) => (config.lint?.rules || []).flat()
-
-export const lint = async (config: Config, env: EnvironmentFunctions) => {
-	const { referenceLanguage, languages, readResources } = config
-
-	const lintRules = getLintRulesFromConfig(config)
-	if (!lintRules.length) {
+export const lint = async (args: {
+	config: Pick<Config, "lint" | "languages" | "referenceLanguage">
+	resources: Resource[]
```

Comment:
The problem with passing the `resources` directly into the `lint` function is, that the function is no longer pure. The sideeffect of that function is, that it mutates the `resources` directly. We could call `strucuredClone` somewhere to make a copy of the object.

Reply:
Good catch! 

I was already wondering if lint mutates the resources or not. But forgot to write a test. I will add a structured clone.